### ISSUE TITLE
Minimum PDF::TextBlock version?

### ISSUE THREAD

**Issue by eserte** [ctrlo/CtrlO-PDF#9]
On some of my smoker systems the test suite fails like this:
```
PERL_DL_NONLAZY=1 "/usr/perl5.20.0/bin/perl" "-MExtUtils::Command::MM" "-MTest::Harness" "-e" "undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')" t/*.t
t/001_base.t ... ok
Ambiguous use of -pi resolved as -&pi() at /usr/perl5.20.0/lib/site_perl/5.20.0/PDF/Builder/Content.pm line 2063.
pdf attribute (a PDF::API2 object) required at /usr/home/cpansand/.cpan/build/2021012611/CtrlO-PDF-0.09-0/blib/lib/CtrlO/PDF.pm line 560.
# Looks like your test exited with 255 before it could output anything.
t/002_basic.t .. 
Dubious, test returned 255 (wstat 65280, 0xff00)
Failed 1/1 subtests 
```

Statitical analysis suggests that this happens if PDF::TextBlock is too old (probably 0.13 is required):
```
****************************************************************
Regression 'mod:PDF::TextBlock'
****************************************************************
Name           	       Theta	      StdErr	 T-stat
[0='const']    	      0.0000	      0.0000	   6.51
[1='eq_0.09']  	     -0.0000	      0.0000	  -5.15
[2='eq_0.10']  	     -0.0000	      0.0000	  -4.60
[3='eq_0.11']  	     -0.0000	      0.0000	  -5.64
[4='eq_0.13']  	      1.0000	      0.0000	57486084140431640.00

R^2= 1.000, N= 94, K= 5
****************************************************************
```

**Comment by abeverley** [comment]
Thanks @eserte you are correct. I'll aim to get a fix released tomorrow. Thanks for reporting.

**Comment by PhilterPaper** [comment]
I am puzzled why Content.pm would blow up there. **pi** is a constant from Math::Trig, not a function call, and here is being used in the bspline() method (drawing Bezier splines). I have no idea what CtrlO::PDF or PDF::TextBlock would be doing drawing splines (especially in a t-test). It's probably not _running_ that code, but failing during _compile_, which has nothing to do with CtrlO::PDF or PDF::TextBlock. Anyway, it's unlikely that requiring the latest PDF::TextBlock version in CtrlO::PDF will have any effect, although it certainly wouldn't _hurt_. My guess is that it's more likely that there's something wrong with your copy of Math::Trig. In any case, I'll continue to monitor this ticket.

**Add:** Perl 5.20 is the minimum acceptable for PDF::Builder, although it will likely (and has) run on earlier versions. It's just that no Perl older than about 6 years is officially supported, so we don't waste time trying to fix problems due to old Perls. Anyway, if your Math::Trig is somehow corrupted (you might try running PDF::Builder's examples/Bspline.pl to exercise that code, if PDF::Builder successfully installed), it might be time to think about upgrading to a more current Perl.

**Comment by eserte** [comment]
Probably I picked not the best fail report --- the pi thing is just a warning, probably with older Math::Trig, but is not causing the failure. Please look here: http://matrix.cpantesters.org/?dist=CtrlO-PDF-0.09 --- the failures happen even with new perl versions.

**Comment by PhilterPaper** [comment]
PDF.pm line 560 is a call to PDF::TextBlock, so it might indeed be useful to require 0.13 or higher of PDF::TextBlock. It's worth a try. I think it must be something about the environment found at these particular test systems, as many other tests with the same Perl version, etc. run fine. Note that PDF::API2 and/or PDF::Builder require current levels, but CtrlO::PDF, as noted, has no minimum requirement for PDF::TextBlock. I probably should have added it while I was in there.

**Comment by abeverley** [comment]
I've just released v0.10 now with PDF::Builder version 0.13 required. Let's see if that fixes it.

**Comment by abeverley** [comment]
All tests now seem to be passing.
